### PR TITLE
Rdm 1964 - Conditional Hide/Show Tabs & TabFields while viewing case

### PR DESCRIPTION
**Pull request by tzarsmango**
https://tools.hmcts.net/jira/browse/RDM-1831

This one is adding conditional hide/show for the CaseViewTabs and Fields on the tabs

The case can be tested using the stubs-unit application

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/app/cases/viewer/case-viewer.component.ts [hunk 1]
```diff
@@ -43,13 +44,13 @@ export class CaseViewerComponent implements OnInit, OnDestroy {
...
 
-    // Clone and sort fields array
-    this.sortedTabs = this.sortedTabs.map(tab => Object.assign({}, tab, {
-      fields: this.orderService.sort(tab.fields)
-    }));
-
     this.caseFields = this.getTabFields();
 
+    // Clone and sort fields array
+    this.sortedTabs = this.sortedTabs
```
Comment (andrewfolga) [inline src/app/cases/viewer/case-viewer.component.ts]: Can we extract the right side of assignment to a well named method? Maybe the comment could be removed as well.

File: src/app/cases/viewer/case-viewer.component.ts [hunk 2]
```diff
@@ -43,13 +44,13 @@ export class CaseViewerComponent implements OnInit, OnDestroy {
...
+      .map(tab => Object.assign({}, tab, { fields: this.orderService.sort(tab.fields) }))
```
Comment (mario-paniccia) [inline src/app/cases/viewer/case-viewer.component.ts]: We have  a field utils class where we are adding utility method for fields. Things like cloning ecc. Can we reuse the clone method in there or move this 'cloning' method there?

File: src/app/shared/conditional-show/conditional-show.model.ts [hunk 3]
```diff
@@ -30,4 +32,18 @@ export class ShowCondition {
   private removeStarChar(s: string) {
     return s.substring(0, s.length - 1);
   }
+
+  matchByCaseFields(caseFields: CaseField[]): boolean {
```
Comment (mario-paniccia) [inline src/app/shared/conditional-show/conditional-show.model.ts]: I think this whole method can be moved in FieldsUtils. 
Also, perhaps we could reuse some method already there to achieve the same result

File: src/app/shared/conditional-show/conditional-show.model.ts [hunk 4]
```diff
@@ -30,4 +32,18 @@ export class ShowCondition {
...
+    return this.match(this.buildState(caseFields));
+  }
+
+  private buildState(caseFields: CaseField[]): any {
```
Comment (mario-paniccia) [inline src/app/shared/conditional-show/conditional-show.model.ts]: I think this whole method can be moved in FieldsUtils. Can we called toValuesMap or similar?
Also, perhaps we could reuse some method already there to achieve the same result

File: src/app/cases/viewer/case-viewer.component.ts (hunk 1 above)
Comment (tzarsmango) [reply src/app/cases/viewer/case-viewer.component.ts]: 👍

File: src/app/cases/viewer/case-viewer.component.ts (hunk 2 above)
Comment (tzarsmango) [reply src/app/cases/viewer/case-viewer.component.ts]: Field utils generally has methods for merging caseFields and formFields, but we're having a two level data being merged & sorted for caseTabs and caseTabsFields, so I prefer this one to stay here

**Pull request merged**